dotnet/src/Connectors/Connectors.Memory.AzureCognitiveSearch/AzureCognitiveSearchMemoryStore.cs

```diff
@@ -428,5 +429,15 @@ private static T RunMemoryStoreOperation<T>(Func<T> operation)
         }
     }
 
+    private static double DistanceToSimilarity(double distance)
+    {
+        return 1 - (1 - distance) / distance;
+    }
+
+    private static double SimilarityToDistance(double similarity)
+    {
+        return 1 / (2 - similarity);
+    }
```

Comment:
I'm not familiar with the formulas but, assuming possible similarity values in the range from 0 to 1, the possible distances given by SimilarityToDistance() range between 0.5 to 1.

Also, if distance ranges from 0 to 1 (does it?), then the similarities given by DistanceToSimilarity() range from 1 to infinity.  However, we know similarities range from 0 to 1, so this methid is not correct!

Isn't the relationship between distance and similarity simply CosineDistance = 1 - CosineSimilarity???

Reply:
the `DistanceToSimilarity` to formula was shared by the Search team. This one is just the inverse of the other formula. I've seen other definitions of Distance, e.g. in StackOverflow, but these are specific to Azure Cognitive Search.

Reply:
Curious to know more.  We can chat in person.  The domain/range for these equations I'm getting is (-infinity, 2) for similarity, and (0, infinity) for distance.

Reply:
happy to chat offline, I think here we're good to merge. 

This formula is just the inverse of the other one. 
**Similarity goes from 0 to 1**, hence:

min: `1 / (2 - 0)` => `0.5`
max: `1 / (2 - 1)` => `1.0`

Reply:
If there are restrictions on the input domain, why does the code not enforce that?